### ISSUE TITLE
multi-class classification

### ISSUE THREAD

**Issue by drag05** [WinVector/vtreat#20]
Hello, I wonder when it would be possible to use vtreat for multi-class / multi-label classification data.
Thank you!

**Comment by JohnMount** [comment]
Very good question.  Thanks for sharing that point.

For the current version of `vtreat` the answer is unfortunately: as a direct step, no.

One could try to work overt `vtreat` to approximate such an effect.  Very roughly the idea would be to step through all class targets and call either `vtreat::designTreatmentsC()` or `vtreat::mkCrossFrameCExperiment()` selecting on the y-aware treatments (`catB` and so-on).  Then `cbind` all of the produced columns together (with column-names disambiguated).  In this mode we are simulating a bunch of "1 versus rest" column treatments.  For the non y-aware treatments we can get the conversion once through `vtreat::desisgnTreatmensZ()`.

I'll leave this issue open until I create a vignette explaining how to do this.

**Comment by drag05** [comment]
Dr. Mount,

Thank you for the prompt replay! 
I find the second suggested option (**designTreatmentsZ()**) more compact as I am dealing with a (very unbalanced) ten-class classification problem. 

I was considering trying this option, as well as I was considering turning the problem into a regression problem and then selecting the probabilities associated with the integer output values only.  

I am waiting for the new vignette with much anticipation as much as I hoping to see a full textbook on data conditioning; being able to employ an organized and documented data conditioning framework on real-life quality data is a pure bliss! 

Thank you!

**Comment by JohnMount** [comment]
Here is the new vignette  [https://winvector.github.io/vtreat/articles/MultiClassVtreat.html](https://winvector.github.io/vtreat/articles/MultiClassVtreat.html).  I have to leave it fairly terse as I don't have a lot of free time this month.  It won't be included in the `CRAN` version of `vtreat` for about a month (need to wait on updating a bit), but the new code can be used on top of the `CRAN` version.

If you have issues or ideas please do either open a new issue or reach out directly ( jmount@win-vector.com ).  We want `vtreat` to be the best way to prepare data for predictive analytics, and I appreciate your help in making it so.

You are right `designTreatmentsZ()` is much more compact. Perhaps one only want to try the mult-steps on a subset of variables (or none at all).

**Comment by drag05** [comment]
Great, thank you!

**Comment by JohnMount** [comment]
I forgot to mention: our formal documentation is [here](https://github.com/WinVector/vtreat/blob/master/extras/vtreat.pdf).  Not quite as long as a text book yet.

**Comment by drag05** [comment]
Dr. Mount,

Thank you - very much appreciated!

I have read almost everything that Dr. Zumel and yourself published on vtreat
and data science - this article having been my very first read on vtreat.
 Initially I hoped binary classification would be treated as a particular case
of multi-class classification which - I find - is given less attention in
general. 

---------- Original Message ----------
